### DISCUSSION TITLE
Best practice to implement authentication and authorization

### DISCUSSION THREAD

**Discussion by Crhis35** [Urigo/graphql-modules#1649] [Q&A] [answered]
Hi i'm Crhis, i'm recently found this package and Im excited to learn I was reading some docs about and those are on legacy version. I'm trying to implement and authentication using injectors, scopes and others but I get stuck if some could help me this is my repo https://github.com/Crhis35/EvoStore in build this project by myself to learn graphql modules with subscription and files uploads and using react on the frontend but for now I just have a basic setup with just one schema (user)

**Comment by victororlyk** [comment, marked as answer]
I also would like to know how we can implement custom directives with it and new version of library
